cpu/atmega2560/periph/gpio.c

```diff
@@ -53,23 +260,188 @@ int gpio_init_int(gpio_t dev, gpio_pp_t pushpull, gpio_flank_t flank, gpio_cb_t
 
 int gpio_read(gpio_t dev)
 {
-    return -1;
+    uint8_t pin;
+
+    if (dev > GPIO_NUMOF)
+    {
+        return -1;
+    }
+
+    pin = gpio_pin_map[dev];
+
+    switch (dev)
+    {
+        case GPIO_0:
+            if ( GPIO_0_STATE & (1 << pin) )
```

Comment:
I assume the only "possible" values you can read from this registers will be `0` and `1`. Why don't you just `return` the read value? You will save many lines of `if else`